### PR TITLE
Adds README-Mandrel.md

### LINKED ISSUE #25: Update readme to provide Mandrel information
Add:
* communication channels
* what is Mandrel and how it compares to Graal
* other?

### PR DESCRIPTION
**Pull request by zakkak**
Closes #25 #32 #33

### CODE REVIEW

File: README-Mandrel.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Mandrel
+
+Mandrel is a downstream distribution of the GraalVM community edition.
+Mandrel's main goal is to provide a `native-image` in-place replacement to be used by [Quarkus](https://quarkus.io).
```
Comment (zakkak) [inline README-Mandrel.md:4]: @maxandersen @adinn please review ^^

File: README-Mandrel.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## How Does Mandrel Differ From Graal
+
+Mandrel releases may differ from the corresponding Graal releases in the following ways:
+  * Mandrel releases may include backports from Graal master that the corresponding Graal release doesn't include
+  * Mandrel may include features and/or fixes that are not yet available on Graal upstream
+  * Mandrel is built using upstream OpenJDK, while Graal is built using labs-OpenJDK
```
Comment (zakkak) [inline README-Mandrel.md:11]: @maxandersen @adinn please review ^^

File: README-Mandrel.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Mandrel
+
+PLACEHOLDER FOR SHORT DESCRIPTION
+
+## How Does Mandrel Differ From Graal
+
+PLACEHOLDER
```
[the remark this answers is not in the document]
Comment (adinn) [reply README-Mandrel.md]: How about this:

Mandrel releases are built from a code base derived from the upstream GraalVM code base, with only very minor changes but some significant exclusions. They support the same native image capability as GraalVM with no significant changes to functionality. They do not include support for Polyglot programming via the Truffle interpreter and compiler framework. In consequence, it is not possible to extend Mandrel by downloading languages from the Truffle language catalogue.

Mandrel is also built slightly differently to GraalVM, using the standard OpenJDK project release of jdk11u.  This means it does not profit from a few small enhancements that Oracle have added to version of OpenJDK used to build their own GraalVM downloads. Most of these enhancements are to the JVMCI module that allows the Graal compiler to be run inside OpenJDK. The others are small cosmetic changes to behaviour. These enhancements may in some cases cause minor differences in the progress of native image generation. They should not cause the resulting images themselves to execute in a noticeably different manner.

File: README-Mandrel.md (hunk 2 above)
Comment (adinn) [reply README-Mandrel.md:11]: How about this:

Mandrel releases are built from a code base derived from the upstream GraalVM code base, with only very minor changes but some significant exclusions. They support the same native image capability as GraalVM with no significant changes to functionality. They do not include support for Polyglot programming via the Truffle interpreter and compiler framework. In consequence, it is not possible to extend Mandrel by downloading languages from the Truffle language catalogue.

Mandrel is also built slightly differently to GraalVM, using the standard OpenJDK project release of jdk11u. This means it does not profit from a few small enhancements that Oracle have added to version of OpenJDK used to build their own GraalVM downloads. Most of these enhancements are to the JVMCI module that allows the Graal compiler to be run inside OpenJDK. The others are small cosmetic changes to behaviour. These enhancements may in some cases cause minor differences in the progress of native image generation. They should not cause the resulting images themselves to execute in a noticeably different manner.

Comment (zakkak) [reply README-Mandrel.md]: Integrated ^^

File: README-Mandrel.md (hunk 1 above)
Comment (adinn) [reply README-Mandrel.md]: I'm really not keen on saying mandrel replaces Quarkus or even just some part of Quarkus. I don't think Oracle will be happy with that idea either.

Comment (adinn) [reply README-Mandrel.md]: I would prefer if Thomas Qvarnstrom decided what to say here.

Comment (zakkak) [reply README-Mandrel.md]: @adinn I guess you mean GraalVM instead of Quarkus. Thanks for your comments.
Summoning @tqvarnst for a review :)

Comment (tqvarnst) [reply README-Mandrel.md]: I agree with @adinn here. Maybe we could say something like this instead: "Mandrel's main goal is to align the `native-image` capabilities from GraalVM with OpenJDK and Red Hat Enterprise Linux libraries to improve maintainability for native Quarkus applications."

Comment (adinn) [reply README-Mandrel.md]: Can we just change this to say "Mandrel's main goal is to provide a `native-image` release specifically to support [Quarkus](https://quarkus.io)."

Comment (adinn) [reply README-Mandrel.md]: Of course, that doesn't explain *why* we want a different release to GraalVM and doesn't explain that we are happy for Quarkus to be used with GraalVM releases.

Comment (adinn) [reply README-Mandrel.md]: Ah, didn't see Thomas's comment

"Mandrel's main goal is to provide a native-image release specifically to support Quarkus. The aim is to align the native-image capabilities from GraalVM with OpenJDK and Red Hat Enterprise Linux libraries to improve maintainability for native Quarkus applications."

Comment (adinn) [reply README-Mandrel.md]: How does that sound @tqvarnst?

Comment (tqvarnst) [reply README-Mandrel.md]: Yes, that sounds good.

Comment (adinn) [reply README-Mandrel.md]: Ok, we have a wrap!

**Comment by adinn** [comment]
@zakkak With Thomas's agreement I think this is reviewed

File: README-Mandrel.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+* libffi-devel
+
+On Fedora/CentOS/RHEL they can be installed with:
+```bash
+dnf install glibc-devel zlib-devel gcc libffi-devel
+```
+
+### Building Mandrel From Source
+
+For building Mandrel from source please see [mandrel-packaging](https://github.com/graalvm/mandrel-packaging)
```
Comment (Karm) [inline README-Mandrel.md:47]: @galderz @zakkak This should eventually become true, but it is not ATTOW.  Let's leave it here. Someone has to do the migration though :)

Comment (zakkak) [reply README-Mandrel.md:47]: Started discussion in https://oss.oracle.com/pipermail/graalvm-dev/2020-June/000061.html

**Pull request merged**